Proposed changes to mushr_cad to add lighting assemblies to platform

pfliegster opened the pull request:
Created changes to the mushr_cad database on branch 'add-lighting-system'  to support the addition of a lighting system to the MuSHR robotic vehicle platform based on Adafruit NeoPixel RGBW LED Sticks and Trinket-M0 MCU for control. I'd love to have you consider incorporating these ideas into the main MUSHR platform and specifically the **mushr_cad** repository. If this is not something you would like to implement in the main project, I'd be interested in finding another way to contribute or highlight these ideas for others to enjoy and learn from as well.

This change mostly affects the **racecar_front_cover_center.scad** design file, but a few other files are affected as well. I've also added STL files to reflect updated model part implementing the proposed changes.

The proposed lighting system relies on two external repos I've created specifically to aid in implementing this lighting system on MuSHR in a modular way (while being usable for other projects or stand-alone).

A very detailed description of the changes in this PR, along with notes on the overall design of a lighting system for MuSHR is included in a separate repo: [pfliegster/mushr-lighting-system](https://github.com/pfliegster/mushr-lighting-system).

Thanks for considering these modifications.
@planc509 @schmittlema

pfliegster committed:
Lighting changes added to forked repo for PR consideration

Pulled in changes to the mushr_cad database on branch 'add-lighting-system'  to support the addition of a lighting system to the MuSHR robotic vehicle platform based on Adafruit NeoPixel RGBW LED Sticks and Trinket-M0 MCU for control. Match documented changes in separate repo [pfliegster/mushr-lighting-system](https://github.com/pfliegster/mushr-lighting-system), except for change #3 which is pending.

pfliegster commented:
Hello, I see that there are lots more changes indicated than are actually present in the proposed PR (and from what showed up for me on my Mac using GitHub Desktop), perhaps EOL differences or whitespace? There are really only a handful of substantive changes to the files, in addition to pulling in referencing external modules (where the bulk of the new design resides). If needed I'll try to figure out why and resubmit if this lighting system concept is interesting to the team. Thanks.

palanc commented:
Hi @pfliegster - This is really awesome, thank you for contributing! Sorry for the long delay in responding.  While adding lighting is really cool, some MuSHR users might not want to spend the extra money/work to use it. So I would be open to including these changes if they were disabled by default. Based off of a quick scan, it looks like you have boolean variables to do this, but I would want them all to be false by default. Or another option could be to have these changes enabled by default, but commit them to a non-master branch. If either of those options would be something you're open to, it would help me a lot with the review process if you could resolve the whitespace/eol diffs that you mentioned in your previous comment.

pfliegster commented:
Excellent, @planc509 , thanks for the feedback on this. I have no problem making the lighting system changes disabled by default. I will spend some time with this, and trying to figure out the other eol/whitespace issue and update as suggested. It may take me a couple of weeks, so I will re-sync to the main branch again if necessary for other changes your team may make in the meantime.

pfliegster committed:
attempt fix EOL. characters (Unix -> Windows)

eliminate large number of whitespace diffs (due to EOL character mismatches) for PR to mushr_cad master.

pfliegster committed:
Fix whitespace issues for pull request

Fix remaining EOL differences (unix -> Windows) to resolve large number of whitespace differences in github pull request.

pfliegster committed:
Disable lighting system components by default

Per recommendation by @planc509, I've disabled the lighting system components by default. If the MuSHR user/builder would like to enable the headlight mount and/or lighting controller, they are independently enabled using two boolean values ... just set these to true.

pfliegster commented:
@planc509 - OK, I have solved the whitespace difference issues on the files in question (sorry about that, but now I know for the future :-) ). I have also disabled the lighting additions by default per your suggestion. Please review again and give it a whirl. Let me know if there are some other changes you'd like to see before accepting the PR.

BTW: I am working on releasing the software components soon & would like to create a tutorial for your webpage as well detailing the HW, SW and modifications that are possible with these lighting system changes. Thank you.
